src/twisted/test/test_sslverify.py

```diff
@@ -1229,43 +1229,61 @@ def test_tlsProtocolsTLSv1Point2Only(self):
         self.assertEqual(options, ctx._options & options)
 
 
+    def test_tlsProtocolsTLSv1Point3Only(self):
+        """
+        When calling L{sslverify.OpenSSLCertificateOptions} with
+        C{insecurelyLowerMinimumTo} and C{lowerMaximumSecurityTo} set to v1.3,
+        it will exclude all others.
+        """
+        opts = sslverify.OpenSSLCertificateOptions(
+            privateKey=self.sKey,
+            certificate=self.sCert,
+            insecurelyLowerMinimumTo=sslverify.TLSVersion.TLSv1_3,
+            lowerMaximumSecurityTo=sslverify.TLSVersion.TLSv1_3,
+        )
+        opts._contextFactory = FakeContext
+        ctx = opts.getContext()
+        options = (SSL.OP_NO_SSLv2 | SSL.OP_NO_COMPRESSION |
+                   SSL.OP_CIPHER_SERVER_PREFERENCE | SSL.OP_NO_SSLv3 |
+                   SSL.OP_NO_TLSv1 | SSL.OP_NO_TLSv1_1 | SSL.OP_NO_TLSv1_2)
+        self.assertEqual(options, ctx._options & options)
+
+
     def test_tlsProtocolsAllModernTLS(self):
         """
         When calling L{sslverify.OpenSSLCertificateOptions} with
         C{insecurelyLowerMinimumTo} set to TLSv1.0 and
-        C{lowerMaximumSecurityTo} to TLSv1.2, it will exclude both SSLs and
-        the (unreleased) TLSv1.3.
+        C{lowerMaximumSecurityTo} to TLSv1.3, it will exclude SSLs.
         """
         opts = sslverify.OpenSSLCertificateOptions(
             privateKey=self.sKey,
             certificate=self.sCert,
             insecurelyLowerMinimumTo=sslverify.TLSVersion.TLSv1_0,
-            lowerMaximumSecurityTo=sslverify.TLSVersion.TLSv1_2,
+            lowerMaximumSecurityTo=sslverify.TLSVersion.TLSv1_3,
         )
         opts._contextFactory = FakeContext
         ctx = opts.getContext()
         options = (SSL.OP_NO_SSLv2 | SSL.OP_NO_COMPRESSION |
-                   SSL.OP_CIPHER_SERVER_PREFERENCE | SSL.OP_NO_SSLv3 |
-                   opts._OP_NO_TLSv1_3)
```

Comment:
This actually *removes* test coverage; we want to ensure that if you intentionally disable TLSv1.3 by lowering the security *below* 1.3, that OP_NO_TLSv1_3 is indeed passed.